Universal Object ID

johnspackman opened the pull request:
See https://github.com/qooxdoo/qooxdoo/issues/9422

johnspackman committed:
add support for async events, where an event handler can return a Promise and cause subsequent actions (including further events) to be delayed until the original event has completed

johnspackman committed:
Provides Universal Widget ID implementation discussed in https://github.com/qooxdoo/qooxdoo/issues/9422

johnspackman committed:
lint & better comments

derrell commented on framework/source/class/qx/core/MObjectId.js:
```diff
@@ -0,0 +1,231 @@
+     * @param id {String|Object} the ID of the object to discard, or the object itself
+     */
+    discardOwnedObject: function(id) {
+      if (!this.__ownedObjects) {
```
What do you think about having this stuff done in a private function, e.g., `__discardOwnedObject`, that also has a `force` argument? Then you could have `addOwnedObject` call `__discardOwnedObject` to discard it from a prior owner. The `force` argument, if `true` would ignore `obj.getOwner()` returning null, and allow the `__changingOwner` member to potentially be set to true.

`discardOwnedObject` would call `__discardOwnedObject` with the `foce` value being false.

The goal, if not obvious, is to allow *changing* ownership with a call to `addOwnedObject` just like a widget can be moved from one container to another by simply calling `add()`.

johnspackman replied:
I did start coding something like that, but because of the additional complication of making sure that IDs are unique within owners when changing owner and/or ID, the processes involved started to get quite complex.  

As the pattern here is to create objects on demand, so there has to be a public `discard` method to undo the on-demand `_createObjectImpl`; also, conceptually it seems that owned objects will normally remain in one owner anyway, so reverting to a situation where it has to be manually removed before adding to another owner seemed acceptable.

On reflection though, I think a couple of points:
* rename `discardOwnedObject` to `removeOwnedObject` for better consistency with the add/remove
* `removeOwnedObject` to allow silently ignore IDs that don't exist in case they have just not been created yet
* and Ill make `removeOwnedObject` remove it from it's current owner

johnspackman committed:
rename `discardOwnedObject` and allowed `addOwnedObject` to automatically remove from the previous owner

cajus commented on framework/source/class/qx/test/core/ObjectId.js:
```diff
@@ -0,0 +1,68 @@
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+ * Fabian Jakobs (fjakobs)
```
Really? ;-)

johnspackman replied:
oops, that Fabian gets everywhere ;)

cajus commented:
It would be nice to get rid of additional code in final builds. Or just make it *dead* using `qx.Environment.select()` when defining the methods.

johnspackman committed:
Remove DOM in non-debug builds, fix comment

johnspackman commented:
>It would be nice to get rid of additional code in final builds. Or just make it dead using qx.Environment.select() when defining the methods.

Do you mean the DOM attributes?  I've just done that, but can't see any other candidates - the `getObject` API is intended for accessing (and overriding) objects as a general use case, and which adds IDs and a path based selection mechanism as a bonus rather than just for testing

cajus commented:
@johnspackman I mean that adding stuff in `qx.core.Object` is included in every qx object. Which might make it bigger and the application slower. Although I don't know how much the impact will be in this case. We solved that here by installing a monkey patch early. In the customer relevant built, the monkey patch is not installed.

johnspackman commented:
In this case the physical impact on applications is quite low - there won't be an increase in memory, unless it's used in which case the map will cause an increase (although presumably not that big, and in any case it is being used for a reason), and there won't be a performance issue either because although the mixin is added to `qx.core.Object` it doesn't add any code except where it is called.

If this was used for just testing then I'd agree it should be removed in production, but IMHO having Universal IDs like this is really very useful when developing applications because it allows the construction of widgets and other objects to be broken out and overridden.

WRT to testing, having a mechanism for IDs baked into the framework will encourage IDs to be used from the outset, which means that when it comes to adding Selenium unit tests the IDs will already be there.  It's going to hurt to have to add IDs retrospectively so there is an incentive to promote IDs as part of day-to-day coding.

I've just added a comment https://github.com/qooxdoo/qooxdoo/issues/9422#issuecomment-346301649 which tries to explain further what I mean about how useful IDs can be in the general case; this is based on my own mixin for IDs (I just checked and found that I have 655 classes which use this pattern already!)

level420 reviewed (changes requested):
Just noticed that lint is throwing some errors:
```
Running lint...

  - Warning: qx.core.MObjectId (166, 9): Loop or condition statement without a block as body

  - Warning: qx.core.MObjectId (168, 9): Loop or condition statement without a block as body

Lint warnings are treated as errors! Please fix!
```

level420 commented:
@johnspackman could you please have a look into my change request? thank you.

johnspackman commented:
@level420 yes - I've kind of put this on hold temporarily because I need to do some tests using the solution that @cajus suggested, because if it works then this PR would be significantly reduced.

johnspackman committed:
Merge pull request #9480 from cboulanger/cboulanger-fix-databinding-errormsg

Improved error messages when databinding fails on missing target setter

level420 committed:
Fix for issue #9481: firefox v59 accidentally accessing IE10 workaround

See https://github.com/qooxdoo/qooxdoo/issues/9481

schnepe2 committed:
added translation job sub-keys (and documentation)

johnspackman committed:
Merge pull request #9486 from schnepe2/generator_enhancement

'translation' generator job enhancement(s)

hkollmann committed:
Merge pull request #9483 from level420/level420-fix-issue-9481

Fix for issue #9481: firefox v59 accidentally accessing IE10 workaround

johnspackman committed:
fix invalid return

hkollmann committed:
Merge pull request #9491 from johnspackman/event-promises-4

Promise support in Events
5 aprovals, 8 days
Thanks @johnspackman

level420 committed:
Fixes #9488: Dispose the previous layout in qx.ui.form.List on orientation change

level420 committed:
fix unintentiol method delition

hkollmann committed:
change hardcoded icon theme Tango to ${qx.icontheme}

hkollmann committed:
add
"asset-let" :
 {
    "qx.icontheme" : ["Tango"]
  }

hkollmann committed:
fix some lint warnings

hkollmann committed:
fix lint command

hkollmann committed:
Merge pull request #9493 from level420/level420-qx-ui-form-list-old-layout-disposal

Fixes #9400 88: Dispose the previous layout in qx.ui.form.List on orientation change

cajus committed:
Merge pull request #9495 from hkollmann/change_tango_to_theme

Change hard coded icon theme Tango to ${qx.icontheme} in framework.

cajus committed:
Merge pull request #9496 from hkollmann/fix_lint_warnings

Fix some lint warnings.

derrell committed:
oo loader template could fail to initialize userAgent (#9492)

* oo loader template could fail to initialize userAgent

In the React Native environment, window.navigator is defined, but
window.navigator.userAgent is not. This patch ensures that userAgent gets
set even if window.navigator was previously set. For consistency, it also
sets window.navigator.product and window.navigator.cpuClass if either is
unset, regardless of whether only those individual members were unset or if
window.navigator was originally unset.

* fix a typo

level420 committed:
Remove arguments.callee in qx.ui.table.model.Simple

and introduce a new third parameter columIndex to the sort functions

level420 committed:
doc error removed

level420 committed:
adapt table demo to new sorting function signature

level420 committed:
adapt table demo to new sorting function signature

level420 committed:
add third parameter to the wrapper in case of single function argument for the comparator

level420 committed:
added tests for custom sorting

level420 committed:
additional documentation as proposed by @derrell

level420 committed:
assure backwards compatibility

and fix a bug where descending sorting did not work if a single function was given as a comparator to setSortMethods.

level420 committed:
added test case for the arguments.callee way

cajus committed:
Merge pull request #9499 from level420/level420-table-model-simple-sorting

Remove arguments.callee in qx.ui.table.model.Simple and introduce a new third parameter columIndex for sort functions

level420 committed:
Remove use of arguments.callee in deprecation warning

This is only a deprecation warning containing arguments.callee for a more accurate indication where the error occurs. Removed in favor of ES5 in strict mode and ES6.

johnspackman committed:
Merge pull request #9500 from level420/level420-remove-arguments-callee-in-table

Remove use of arguments.callee in deprecation warning

johnspackman committed:
Promise support in Events (addition to #9491) (#9498)

* Fix promise behaviour when `qx.promise===false`

fixes https://github.com/qooxdoo/qooxdoo/issues/9494

* Fix issues when qx.promise==false

fixes https://github.com/qooxdoo/qooxdoo/issues/9494

* Fix promise behaviour when `qx.promise===false`

fixes https://github.com/qooxdoo/qooxdoo/issues/9494

* Fix issues when qx.promise==false

fixes https://github.com/qooxdoo/qooxdoo/issues/9494

* mod testrunner to fix promise warnings in test app

* unnecessary `var`

* fix broken tests in `qx.test.Promise`

level420 committed:
Fix non working non strict mode compatibility in qx.ui.table.model.Simple

We don't need to take columnIndex from arguments.callee, as we get it anyway from wrapper around the sort function created in method sortByColumn, as the third parameter! It is enough for compatibility to make arguments.callee.columnIndex available in the user supplied sort function.

oetiker committed:
Merge pull request #9505 from level420/level420-table-model-simple-no-arguments-callee

Fix non working non strict mode compatibility fix in qx.ui.table.model.Simple

REMqb committed:
~ Fix Bug when breaking target property chain #9506

hkollmann committed:
Merge pull request #9507 from RemiHeugue/fix-breaking-target-property-chain

5 approvals, 8 days, merged.
Thanks @RemiHeugue!

milandamen committed:
[New] allow drop-downs to be wider than their parent

milandamen committed:
Improved jsDoc documentation

milandamen committed:
Improved PR based on comments

level420 committed:
Disable embedded audio/video event test on travis and fix macro definitions for travis generator run

level420 committed:
scale 20 may lead to the travis unit test to run into a timeout. downscaling to 4

oetiker committed:
allow to set font weight

milandamen committed:
Improved __defaultModel jsDoc

hkollmann committed:
Merge pull request #9514 from level420/fix-travis-run-parameters

5 days, 5 approvals. merged.
Thanks, @level420!

level420 committed:
attempt to split the travis testrunner tests into 3 separate tests

level420 committed:
Use stages to separate the test runs

level420 committed:
Separate the coverage stage from the Edge stage

level420 committed:
Really separate the coverage stage from the Edge stage

level420 committed:
add safari and run lint and coverage within one stage

level420 committed:
disable safari again because of failing tests

level420 committed:
Removed @zaucker from CODOWNERS with his agreeement

hkollmann committed:
Merge pull request #9519 from level420/split-travis-tests

Fast merge, thanks @level420!

hkollmann committed:
Merge pull request #9516 from oetiker/font-weight-patch

all have aproved,    thanks @oetiker!

hkollmann committed:
Merge pull request #9512 from milandamen/grow-dropdown

more than 5 days, 7 approvals, merged.
Thanks @milandamen!

level420 committed:
Pseudo doc change to re-trigger a fresh travis run

hkollmann committed:
Merge pull request #9517 from milandamen/improved-documentation

Improved jsDoc documentation
6 approvals, 6 days. Merged.
Thanks alot @milandamen!

cajus committed:
Merge pull request #9520 from level420/codeowners-remove-zaucker

Removed @zaucker from CODEOWNERS with his agreement

peuter committed:
fix padding in font-mode

ronkie committed:
Appearance property "base" works incorrect, if appearance from parent theme has string value. #9429 (#9509)

* Appearance property "base" works incorrect, if appearance from parent theme has string value. #9429
Always resolve alias as parent theme appearance can have a different one

* Appearance property "base" works incorrect, if appearance from parent theme has string value. #9429
Store aliases and styles per theme, as in different themes aliases can point to different appearances

Adolfrizer committed:
Add decoration entries in another order (#9518)

mbgonicus committed:
Remove background image style when switching to icon font

hkollmann committed:
Merge pull request #9522 from peuter/fix-icon-padding

Full aproval. Merged.
Thanks @peuter

level420 committed:
Fix for #9524: fix recursive included style propagation

Replaced the faulty recursive decorator include algorithm.

level420 committed:
Fix a typo

level420 committed:
Removed comment which applied to an older revision

level420 committed:
start recursion with the current decorator, copying styles over into an empty style object

hkollmann committed:
Merge pull request #9523 from mbgonicus/fonticon

5 day, 6 approvals, merged
Thanks @mbgonicus !

level420 committed:
Preserve precedence within the decorator hierarchy and sort properties before applying so that short cut properties like width come before explicit ones like widthLeft

level420 committed:
remove property sorting

level420 committed:
do not return an assignment

johnspackman committed:
Merge pull request #9525 from level420/level420-fix-9524

Fix for #9524: fix recursive included decorator style propagation

mever committed:
[Fixed] {qx.ui.core.MPlacement#getLayoutLocation} fails with windows

oetiker committed:
the property with the `testCharacters` is called `comparisonString`

the confusion seems to come from qooxdoo-compiler ... somehow ... https://github.com/qooxdoo/qooxdoo-compiler/issues/129

oetiker committed:
Create Issues Templates

This is a new github feature ... providing some guidance in creating issues.

the current content of these is just what github proposes ...

hkollmann committed:
Merge pull request #9527 from oetiker/comparions-string-fix

full approval, merged.
Thanks @oetiker!

cajus committed:
Merge pull request #9526 from mever/getLayoutLocation-fix

[Fixed] {qx.ui.core.MPlacement#getLayoutLocation} fails with windows

oetiker committed:
icon fonts are not necessarly square ... take the aspect ratio of the glyphs into account when displaying an iconfont

oetiker committed:
change font size when applying scaled mode

oetiker committed:
as long as width or height is not set explicitly their value will be null, use sizeHint as a fallback

Commit:
Added require qx.module.util.Array since merge function needs it to function

hkollmann committed:
Merge pull request #9529 from qooxdoo/oetiker-issues-5

full quorum, merging, thank you @oetiker !

cajus committed:
Merge pull request #9531 from sqville/sqville_objmerge_fix

Added require qx.module.util.Array since merge function needs it…

johnspackman committed:
Merge pull request #9530 from oetiker/iconfont-sizing-fix

icon fonts are not necessarly square

cajus committed:
Re-add missing package.json

cajus committed:
Removed one more left over

hkollmann committed:
Merge pull request #9532 from cajus/demobrowser-cure

full aproval. Merged. Thanks @cajus!

hkollmann committed:
fix bumpqxversion.py to reflect the deletes of package.json in some apps.

oetiker committed:
Merge pull request #9533 from hkollmann/fix_bumpqxversion.py

fix bumpqxversion.py to reflect the deletes of package.json in some apps

peuter committed:
apply fix from 222450f3aad2e5bd7b7eb7102e3565d916069749 to double renderer (#9534)

milandamen committed:
[Fixed] basic column model disposing cell editor factory when it shouldn't (#9536)

derrell committed:
add qx.Promise.promisify static method

derrell committed:
remove use of `this` in static methods

derrell committed:
fix a typo

milandamen committed:
[New] added File to Xhr whitelist because File inherits from Blob
See https://developer.mozilla.org/en-US/docs/Web/API/File

milandamen committed:
Improved jsDoc documentation

hkollmann committed:
Merge pull request #9539 from derrell/add-promisify

Full approval, merged. Thanks @derrell!

derrell committed:
Add more documentation (and an example) for promisify (#9542)

* add more documentation (and an example) for promisify

* fix typo in example code

* don't use ES6 in example

* fix indentation (ES6 functions vs old-style)

mever committed:
[Fixed] exception when opening the virtual combo box without selection

hkollmann committed:
Merge pull request #9540 from milandamen/xhrFile

> 5 approvals, > Five days. merged,
Thanks, @milandamen!

level420 committed:
preselect the matching item even if there is no current selection

level420 committed:
fixed typo in comment

oetiker committed:
Merge pull request #9543 from mever/minorFix

[Fixed] exception when opening the virtual combo box without selection

johnspackman committed:
Adds support for a special "null" value in `qx.ui.form.SelectBox` and `qx.data.controller.List`

johnspackman committed:
Fixes regression introduced by recent addition of promises to event handler return values

johnspackman committed:
Remove errant `console.log` calls

johnspackman committed:
comments

Commit:
Merge pull request #1 from qooxdoo/master

Catching up local qx with qx master

Commit:
added "boot" : "function" to the __allowedKeys. I then added "config.boot()" to the last line of qx.Theme.define() function with if condition to ensure that it exists

ronkie committed:
https://github.com/qooxdoo/qooxdoo/issues/9551 blur and deactivate widget before activating the blocker

johnspackman committed:
Adds `qx.Promise.allOf`

Commit:
Updated example and spacing per derrell's request

Commit:
fixed example - removed brackets around the function call. Now example matches true implementation and wont break the app

ronkie committed:
https://github.com/qooxdoo/qooxdoo/issues/9551 blur and deactivate widget before activating the blocker
Changed variable declaration into several lines

johnspackman committed:
remove ES6 code

johnspackman committed:
no message

johnspackman committed:
test for non-promise values

johnspackman committed:
documentation change

peuter committed:
use other separator than " for classname because it breaks the generated
code of properties added by mixins.

peuter committed:
add hash function to the interface to make it overridable

peuter committed:
deleted module

schnepe2 committed:
Update Table.js

derrell committed:
allow continuous debugging of non-freed, with periodic printout since epoch

derrell committed:
Ensure no memory leak. Certain stored sender objects could previously cause it

oetiker committed:
getting travis PR builds back (#9562)

oetiker committed:
Merge branch 'master' into message-cleanup

oetiker committed:
Merge branch 'master' into debug-continuous

oetiker committed:
Merge branch 'master' into issue_9500

oetiker committed:
Merge branch 'master' into fix-marshaller

oetiker committed:
Merge branch 'master' into qx-promise-allOf

oetiker committed:
Merge branch 'master' into blocker

oetiker committed:
Merge branch 'master' into theme_prep_function

oetiker committed:
Merge branch 'master' into selectbox-allow-null-2

oetiker committed:
Merge branch 'master' into fix-drag-drop-regression

oetiker committed:
empty change to trigger rebuild

oetiker committed:
dummy edit

oetiker committed:
Merge pull request #9561 from derrell/message-cleanup

Ensure no memory leak. Certain stored objects could previously cause one.

oetiker committed:
Merge branch 'master' into debug-continuous

oetiker committed:
Merge branch 'master' into issue_9500

oetiker committed:
Merge branch 'master' into fix-marshaller

oetiker committed:
Merge branch 'master' into qx-promise-allOf

oetiker committed:
Merge branch 'master' into blocker

oetiker committed:
Merge branch 'master' into theme_prep_function

oetiker committed:
Merge branch 'master' into selectbox-allow-null-2

oetiker committed:
Merge branch 'master' into fix-drag-drop-regression

hkollmann committed:
Merge pull request #9550 from johnspackman/selectbox-allow-null-2

5 approvals, Long time. Thanks @johnspackman !

hkollmann committed:
Merge branch 'master' into theme_prep_function

hkollmann committed:
Merge branch 'master' into blocker

hkollmann committed:
Merge pull request #9554 from sqville/theme_prep_function

6 approvals, 20 days gone. Thanks @sqville!

hkollmann committed:
Merge branch 'master' into blocker

hkollmann committed:
Merge pull request #9555 from ronkie/blocker

5 approvals, 17 days. merged. Thanks @ronkie!

hkollmann committed:
Merge branch 'master' into qx-promise-allOf

hkollmann committed:
Merge pull request #9557 from johnspackman/qx-promise-allOf

5 approvals, time is over, Thanks @johnspackman!

hkollmann committed:
Merge branch 'master' into issue_9500

hkollmann committed:
Merge pull request #9559 from schnepe2/issue_9500

6 approvals, 8 days. Merged. Thanks @schnepe2

hkollmann committed:
Merge branch 'master' into fix-drag-drop-regression

hkollmann committed:
Merge pull request #9549 from johnspackman/fix-drag-drop-regression

full approval, merged. Thanks @johnspackman!

hkollmann committed:
Merge branch 'master' into debug-continuous

oetiker committed:
* make travis run stage scripts in parallel
* add chrome test

oetiker committed:
fix missing blocks (discovered by newly active lint run)

oetiker committed:
nameing the jobs

peuter committed:
fix tests and add the old way as legacy method

peuter committed:
Merge branch 'fix-marshaller' of github.com:peuter/qooxdoo into fix-marshaller

peuter committed:
fix store test

oetiker committed:
Merge pull request #9563 from oetiker/travis-speedup

travis improvements

peuter committed:
Merge branch 'master' into fix-marshaller

hkollmann committed:
Merge branch 'master' into debug-continuous

hkollmann committed:
Merge pull request #9560 from derrell/debug-continuous

5 approvals, 4 days. Merged. Thanks @derrell!

oetiker committed:
Merge branch 'master' into fix-marshaller

schnepe2 committed:
Update Keyboard.js

When unifying the "gecko", "webkit" and "mshtml" specials to behave just as the "webkit" part it works on
(all running on Windows 10 - x64)
- Firefox 61.0.2 (64-Bit) "Quantum"
- IE 11 11.611.16299.0
- Chrome 68.0.3440.106 (64-Bit)
- Edge 41.16299.611.0
- PaleMoon 28.0.0.1 (32-bit)

hkollmann committed:
Merge pull request #9558 from peuter/fix-marshaller

full aproval. merged. Thanks @peuter!

hkollmann committed:
Merge branch 'master' into issue_9323

milandamen committed:
Merge branch 'master' into improved-documentation-2

milandamen committed:
Changed all local class references to be absolute

oetiker committed:
Merge branch 'master' into improved-documentation-2

milandamen committed:
Fixed TestFunction mentions to be absolute path

yybalam committed:
Merge branch 'master' of github.com:qooxdoo/qooxdoo

yybalam committed:
Fix missing interface at mobile form widgets.

johnspackman committed:
Fixes issue #9569 where `modelSelection` does not work properly with `qx.data.controller.List.allowNull`

johnspackman committed:
fix lint warning; improve comments

oetiker committed:
double timeout delays

oetiker committed:
Merge pull request #9571 from oetiker/more-timeout

double timeout delays

level420 committed:
Add flush, some timeout and a longer in order to reliably succeed tests on firefox

The test "test dropdown list wider than selectbox" creates a test string ("asd.....") which in firefox has a width of ~650px where in chrome it has ~670px. therefore the test fails in firefox were the assertion expected a width of greater than 666px. Addes some more charcters to reliably fullfill the expected width. 
Both tests are not running reliable on high loads and in containers because of the virtual box needs some time to create the dom and calculate width etc. therefore a flush was added and the test "test dropdown list same width as selectbox" has now a timout of 10 before the results are evalutated

cajus committed:
Merge branch 'master' into iss9567

oetiker committed:
Merge branch 'master' into level420-fix-ff-test-virtual-box

level420 committed:
Disable test "qx.test.ui.embed Flash:testLoadTimeout" on travis

oetiker committed:
Merge pull request #9572 from level420/level420-fix-ff-test-virtual-box

Add flush, some timeout and a longer in order to reliably succeed VirtualSelectBox tests on firefox

hkollmann committed:
Merge branch 'master' into issue_9323

hkollmann committed:
Merge branch 'master' into iss9567

hkollmann committed:
Merge pull request #9565 from schnepe2/issue_9323

6 approvals, 12 days over. merged. Thanks @schnepe2!

hkollmann committed:
Merge branch 'master' into improved-documentation-2

hkollmann committed:
Merge pull request #9541 from milandamen/improved-documentation-2

6 approvals, merged. Thanks @milandamen!

hkollmann committed:
Merge branch 'master' into iss9567

oetiker committed:
fix compiler breakage for nested if else structures

oetiker committed:
replace if else extravaganza with case

hkollmann committed:
Merge pull request #9568 from yybalam/iss9567

Full approval, Thanks @yybalam!

oetiker committed:
Merge branch 'master' into image-iconfont-build-fix

level420 committed:
Merge branch 'master' into fix-list-controller-null-selection

hkollmann committed:
Merge pull request #9573 from oetiker/image-iconfont-build-fix

full aproval, merged. Thanks,  @oetiker!

level420 committed:
Merge branch 'master' into fix-list-controller-null-selection

hkollmann committed:
Merge pull request #9570 from johnspackman/fix-list-controller-null-selection

5 approvals, more than 5 days. merged. Thanks @johnspackman !

mbgonicus committed:
Throw error on getInstance() if instance is currently being created

mbgonicus committed:
Fixed tests that dispose singletons; fixed typeo

mbgonicus committed:
Comment why $$instance set to null in getInstance

hkollmann committed:
Merge pull request #9575 from mbgonicus/singleton

Full approval, merged. Thanks @mbgonicus !

oetiker committed:
make sure theme colors get applied to iconFonts

The problem is that when the textcolor would get applied, it does not as at that time the source property is not yet set for themed fontIcons.

oetiker committed:
Merge branch 'master' into iconFont-theming-fix

hkollmann committed:
Merge pull request #9577 from oetiker/iconFont-theming-fix

5 days, 5 approvals. merged. Thanks @oetiker !

peuter committed:
use promises only if qx.promise is true

closes #9576

peuter committed:
fix docs

peuter committed:
document whats happening if promises are disabled

peuter committed:
fix blob requests

hkollmann committed:
Merge pull request #9578 from peuter/fix-9576

Full aproval, merged. Thanks @peuter!

hkollmann committed:
Merge branch 'master' into fix-blob

peuter committed:
add getter/setter for the srcObject prorperty of a HTMLMediaElement

hkollmann committed:
Merge pull request #9579 from peuter/fix-blob

6 days, 6 approvals, merged. Thanks @peuter!

level420 committed:
Merge branch 'master' into fix-srcObject

peuter committed:
fire events if the filteredChange differs from the old value (#9580)

level420 committed:
Merge branch 'master' into fix-srcObject

oetiker committed:
killing firefox beta testing on SL

oetiker committed:
Merge pull request #9583 from qooxdoo/oetiker-patch-1

killing firefox beta testing on SL

hkollmann committed:
Merge branch 'master' into fix-srcObject

hkollmann committed:
Merge pull request #9582 from peuter/fix-srcObject

6 approvals, 6 days. merged. Thanks @peuter!

derrell committed:
remove stored function during dispose

johnspackman committed:
Merge pull request #9587 from derrell/timer-mem-leak

remove stored function during dispose

milandamen committed:
Fixed resetValue of VirtualSelectBox not resetting selected item

cajus committed:
Merge pull request #9588 from milandamen/selectbox-reset

Fixed setValue of VirtualSelectBox not resetting selected item

hkollmann committed:
fix: readme.md -> README.md (#9584)

cajus committed:
Remove global invalid tooltip on manager disposal

If a form is currently invalid and the validation manager gets disposed,
it can happen that the tooltip is still visible - even if the form and
the widget which receives the invalid message are already disposed.

Calling _showTooltip(false) (strange naming due to inverse logic, though)
does everything we need to close the tooltip on disposal.

milandamen committed:
Fixed table removeSelectionInterval not correcting selection when removing rows (#9581)

hkollmann committed:
Merge branch 'master' into fix-global-tooltip-removal

hkollmann committed:
Merge pull request #9589 from cajus/fix-global-tooltip-removal

hkollmann committed:
add support for automatic npm deploy to travis build (#9586)

* add support for automatic npm deploy to travis build

* add - #

hkollmann committed:
change file mode to +x

cajus committed:
Merge pull request #9590 from hkollmann/chmod_deploy_npm

change file mode for deploy_npm to +x

hkollmann committed:
the exclude of "qx.event.GlobalError" leads to errors during testing with current master. (generate npm-package-test). (#9585)

The method qx.event.GlobalError.observeMethod is needed in different places.
So i removed the exclude

derrell committed:
Allow excluding the horizontal scrollbar in a metacolumn if not required

derrell committed:
Scroller's horizontalScrollBarVisible property is tri-state to hide vs exclude

derrell committed:
allow non-boolean settings that map to truthy/falsy

derrell committed:
make use of nullable scrollbar visibility property

derrell committed:
horNeeded and verNeeded must be booleans

level420 committed:
Merge branch 'master' into exclude-metacolumn-scrollbar

hkollmann committed:
Did some reworking, found npm error: We need a newer version to use access tokens.
Did some testing with private qooxdoo repos. There the script runs.

oetiker committed:
Merge pull request #9592 from hkollmann/fix_deploy_npm

fix deploy_npm

oetiker committed:
Merge branch 'master' into exclude-metacolumn-scrollbar

hkollmann committed:
Merge pull request #9591 from derrell/exclude-metacolumn-scrollbar

Exclude metacolumn scrollbar

johnspackman committed:
no message

johnspackman committed:
Merge branch 'object-ids-2' of github.com:johnspackman/qooxdoo into object-ids-2

* 'object-ids-2' of github.com:johnspackman/qooxdoo:
  Remove DOM in non-debug builds, fix comment
  rename `discardOwnedObject` and allowed `addOwnedObject` to automatically remove from the previous owner
  lint & better comments
  Provides Universal Widget ID implementation discussed in https://github.com/qooxdoo/qooxdoo/issues/9422

# Conflicts:
#	framework/source/class/qx/core/Id.js
#	framework/source/class/qx/core/MObjectId.js

johnspackman commented:
closing this PR in favour of a new one (coming shortly) due to problems rebasing :(
